app/workers/create_asset_relationship_worker.rb

```diff
@@ -0,0 +1,69 @@
+class CreateAssetRelationshipWorker < WorkerBase
+  def perform(counter_start, counter_end)
+    logger.info("CreateAssetRelationshipWorker start!")
+    assetable_type = AttachmentData
+    assetables = assetable_type.where(id: counter_start..counter_end)
+    logger.info "Number of #{assetable_type.to_s} found: #{assetables.count.to_s}"
+    logger.info "Creating Asset for #{counter_end} records"
+
+    count = 0
+    assetables.each do |assetable|
+      break if counter_start > counter_end
+      assetable.use_non_legacy_endpoints = true;
+      assetable.save
+      begin
```

Comment:
saving the original and the thumbnail in the same begin-rescue is problematic. If the orginal raises an error, the code will never attempt to save the thumbnail